### ISSUE TITLE
Alert mechanism

### ISSUE THREAD

**Issue by coke** [Raku/doc-website#372]
It would be nice to have an alert popup available on the docs site for things like:

* Hey, we added an ebook
* Updated docs to include 2024.04 release
* 6.e dropped!

etc. Be nice if

* We could support at least one message at a time. We do NOT need to support multiples but it if it's easy...
* Once a user has hidden a message it goes away.
* We don't need an archive of messages.
* OK if we need to do a deploy to get a new message (don't need a way for admins to create it dynamically through the site)

**Comment by patrickbkr** [comment]
I like the idea.

> What about accumulated messages?

I say lets display only the last message. If we want to make sure people have a chance to see older ones, we could have an "Announcements" page where the full history can be seen.

**Comment by patrickbkr** [comment]
Sounds good.

**Comment by coke** [comment]
The bulma model is going to be a better experience than `alert()`, IME.

**Comment by finanalyst** [comment]
@coke @dontlaugh @lizmat A PRs to address this issue. It will be in the `docs-dev` branch, so the results will be visible on `https://docs-dev.raku.org`.

An issue to be resolved if the PR meets the requirements of the ask is where to put the 'announcements.rakudoc' page.
For the PR, it is located in `Website/structure-sources` and so is local to this repo.
However, this has the disadvantage that it will not be included in the `Introduction` page (tab). [This is a limitation of `Collection`'s design, which I plan to remove once I start working on the replacement of Collection after RakuAST-RakuDoc is complete]
Since `Announcements` is not listed, it cant be found onsite (of course we could include a link to it in eg About)

My preference would be to have `announcements.rakudoc` as a file in the `Raku/docs` repository immediately inside `doc/`.

**Comment by finanalyst** [comment]
@coke a small `announcements.rakudoc` file needs to be created. Is the test file OK?

**Comment by finanalyst** [comment]
@coke @dontlaugh @lizmat @patrickbkr PR now working on [docs-dev deployment eg introduction](https://docs-dev.raku.org/introduction). 
First time site opens, you  get latest message. Then refresh page, and the modal pane will not fire!
If you want to see all announcements (assuming you dont click on the link in the modal pane), [this is the url](https://docs-dev.raku.org/announcements).
Suppress toggle is at `navigation_bar->More->announcements`
If you want to test with fresh window, then on Firefox (I dont know the sequence for other browsers), goto developers window/tab (on Ubuntu its F12), then 'storage', then 'local storage', then the element labeled 'docs-dev.raku.org', then delete the item labelled 'announcementState'.
See posts above about outstanding issues.

**Comment by finanalyst** [comment]
PR merged to main without an announcements.rakudoc page. I'll raise a PR on Raku/doc  with a short source. Once that is merged into main, the announcements plugin will serve the topmost announcement after the next build here (about 2 hours later).

**Comment by finanalyst** [comment]
closing as completed
